coap-core/src/main/java/com/mbed/coap/packet/CoapResponse.java

```diff
@@ -99,14 +101,23 @@ public String getPayloadString() {
         return payload.toUtf8String();
     }
 
+    @Deprecated
     public SeparateResponse toSeparate(Opaque token, InetSocketAddress peerAddress, TransportContext transContext) {
-        return new SeparateResponse(this, token, peerAddress, transContext);
+        return new SeparateResponse(this.withContext(transContext), token, peerAddress);
     }
 
     public SeparateResponse toSeparate(Opaque token, InetSocketAddress peerAddress) {
         return toSeparate(token, peerAddress, TransportContext.EMPTY);
```

Comment:
If above method is deprecated (which make sense) then this one should use directly constructor.